### PR TITLE
Precision exploration

### PR DESCRIPTION
**Pull request by pablooliveira**
This is a follow-up to @Avaquet work in #179. Now it is in a verificarlo project branch.

### CODE REVIEW

**Comment by pablooliveira** [comment]
@Avaquet I have rebased your PR #179 with @roblim1 fix and current master. I have updated tests after the fix to vprec backed with denormals.

The merge was a bit difficult. Could you please check that I have not messed up and review this PR ?

**Comment by pablooliveira** [comment]
@Avaquet if you need to change things you have write access to verificarlo/precision-exploration.

**Comment by pablooliveira** [comment]
@Avaquet I would like to merge this PR soon. You told me you wanted to push some changes before the merge ?

**Review by Avaquet — APPROVED** [review]

**Pull request merged**
